lib/src/main/java/io/github/taz03/jia/InstagramClient.java

```diff
@@ -103,6 +138,10 @@ public LoginResponse login() throws Exception {
      * @return A LoginResponse object containing information about the user and their session.
      */
     public LoginResponse login(Supplier<String> verificationCodeSupplier) throws Exception {
+        if (userAgent == null || androidId == null){
+            configureDeviceIds();
+        }
```

Comment:
do this in the constructor it self, a method should always only do one thing

Reply:
all data should be assigned on constructor call

Reply:
also it won't be blocking if u don't use url, idk y do that in first place if the file is present locally